pyproject.toml

```diff
@@ -340,7 +340,6 @@ devel-tests = [
     "coverage>=7.2",
     "pytest-asyncio>=0.23.3",
     "pytest-cov>=4.1.0",
-    "pytest-httpx>=0.21.3",
```

Comment:
I've just wondering could we replace `httpx` by something else, e.g. `requests`. Looks like it uses in just couple places.

Seems like we add this as optional dependency when we can't use `requests` due to legal reason:
- https://github.com/apache/airflow/pull/15781
- https://github.com/psf/requests/pull/5797

But seems like it resolved now, at least we use `requests` into the airflow/api_internal/internal_api_call.py

https://github.com/apache/airflow/blob/541c9addb6b2ee56244793503cbf5c218e80dec8/airflow/api_internal/internal_api_call.py#L25

Reply:
Not related to this PR. Just question about potential follow up.

Reply:
Yeah. Quite possibly we could do it. Indeed it took us about 6 months to convince requests maintainers to get rid of chardet and we finally succeeded, so yes we could fully switch to requests now.

Reply:
I guess at least we need to add `requests` in core dependencies, because right now Internal API works because `requests` is a dependency of the dependencies, e.g. Connexion

<details>
  <summary>pipdeptree --packages apache-airflow</summary>
  
```
apache-airflow==2.9.0.dev0
├── alembic [required: >=1.6.3,<2.0, installed: 1.13.1]
│   ├── importlib-metadata [required: Any, installed: 6.11.0]
│   │   └── zipp [required: >=0.5, installed: 3.17.0]
│   ├── importlib-resources [required: Any, installed: 5.13.0]
│   │   └── zipp [required: >=3.1.0, installed: 3.17.0]
│   ├── Mako [required: Any, installed: 1.3.0]
│   │   └── MarkupSafe [required: >=0.9.2, installed: 2.1.3]
│   ├── SQLAlchemy [required: >=1.3.0, installed: 1.4.51]
│   │   └── greenlet [required: !=0.4.17, installed: 3.0.3]
│   └── typing-extensions [required: >=4, installed: 4.9.0]
├── argcomplete [required: >=1.10, installed: 3.2.1]
├── asgiref [required: Any, installed: 3.7.2]
│   └── typing-extensions [required: >=4, installed: 4.9.0]
├── attrs [required: >=22.1.0, installed: 23.2.0]
├── blinker [required: Any, installed: 1.7.0]
├── colorlog [required: >=4.0.2,<5.0, installed: 4.8.0]
├── ConfigUpdater [required: >=3.1.1, installed: 3.2]
├── connexion [required: >=2.10.0,<3.0, installed: 2.14.2]
│   ├── clickclick [required: >=1.2,<21, installed: 20.10.2]
│   │   ├── click [required: >=4.0, installed: 8.1.7]
│   │   └── PyYAML [required: >=3.11, installed: 6.0.1]
│   ├── Flask [required: >=1.0.4,<2.3, installed: 2.2.5]
│   │   ├── click [required: >=8.0, installed: 8.1.7]
│   │   ├── importlib-metadata [required: >=3.6.0, installed: 6.11.0]
│   │   │   └── zipp [required: >=0.5, installed: 3.17.0]
│   │   ├── itsdangerous [required: >=2.0, installed: 2.1.2]
│   │   ├── Jinja2 [required: >=3.0, installed: 3.1.3]
│   │   │   └── MarkupSafe [required: >=2.0, installed: 2.1.3]
│   │   └── Werkzeug [required: >=2.2.2, installed: 2.2.3]
│   │       └── MarkupSafe [required: >=2.1.1, installed: 2.1.3]
│   ├── inflection [required: >=0.3.1,<0.6, installed: 0.5.1]
│   ├── itsdangerous [required: >=0.24, installed: 2.1.2]
│   ├── jsonschema [required: >=2.5.1,<5, installed: 4.21.0]
│   │   ├── attrs [required: >=22.2.0, installed: 23.2.0]
│   │   ├── importlib-resources [required: >=1.4.0, installed: 5.13.0]
│   │   │   └── zipp [required: >=3.1.0, installed: 3.17.0]
│   │   ├── jsonschema-specifications [required: >=2023.03.6, installed: 2023.12.1]
│   │   │   ├── importlib-resources [required: >=1.4.0, installed: 5.13.0]
│   │   │   │   └── zipp [required: >=3.1.0, installed: 3.17.0]
│   │   │   └── referencing [required: >=0.31.0, installed: 0.31.1]
│   │   │       ├── attrs [required: >=22.2.0, installed: 23.2.0]
│   │   │       └── rpds-py [required: >=0.7.0, installed: 0.17.1]
│   │   ├── pkgutil-resolve-name [required: >=1.3.10, installed: 1.3.10]
│   │   ├── referencing [required: >=0.28.4, installed: 0.31.1]
│   │   │   ├── attrs [required: >=22.2.0, installed: 23.2.0]
│   │   │   └── rpds-py [required: >=0.7.0, installed: 0.17.1]
│   │   └── rpds-py [required: >=0.7.1, installed: 0.17.1]
│   ├── packaging [required: >=20, installed: 23.2]
│   ├── PyYAML [required: >=5.1,<7, installed: 6.0.1]
│   ├── requests [required: >=2.9.1,<3, installed: 2.31.0]
│   │   ├── certifi [required: >=2017.4.17, installed: 2023.11.17]
│   │   ├── charset-normalizer [required: >=2,<4, installed: 3.3.2]
│   │   ├── idna [required: >=2.5,<4, installed: 3.6]
│   │   └── urllib3 [required: >=1.21.1,<3, installed: 1.26.18]
│   └── Werkzeug [required: >=1.0,<2.3, installed: 2.2.3]
│       └── MarkupSafe [required: >=2.1.1, installed: 2.1.3]
├── cron-descriptor [required: >=1.2.24, installed: 1.4.0]
├── croniter [required: >=0.3.17, installed: 2.0.1]
│   ├── python-dateutil [required: Any, installed: 2.8.2]
│   │   └── six [required: >=1.5, installed: 1.16.0]
│   └── pytz [required: >2021.1, installed: 2023.3.post1]
├── cryptography [required: >=0.9.3, installed: 41.0.7]
│   └── cffi [required: >=1.12, installed: 1.16.0]
│       └── pycparser [required: Any, installed: 2.21]
├── Deprecated [required: >=1.2.13, installed: 1.2.14]
│   └── wrapt [required: >=1.10,<2, installed: 1.16.0]
├── dill [required: >=0.2.2, installed: 0.3.1.1]
├── Flask [required: >=2.2,<2.3, installed: 2.2.5]
│   ├── click [required: >=8.0, installed: 8.1.7]
│   ├── importlib-metadata [required: >=3.6.0, installed: 6.11.0]
│   │   └── zipp [required: >=0.5, installed: 3.17.0]
│   ├── itsdangerous [required: >=2.0, installed: 2.1.2]
│   ├── Jinja2 [required: >=3.0, installed: 3.1.3]
│   │   └── MarkupSafe [required: >=2.0, installed: 2.1.3]
│   └── Werkzeug [required: >=2.2.2, installed: 2.2.3]
│       └── MarkupSafe [required: >=2.1.1, installed: 2.1.3]
├── Flask-Caching [required: >=1.5.0, installed: 2.1.0]
│   ├── cachelib [required: >=0.9.0,<0.10.0, installed: 0.9.0]
│   └── Flask [required: Any, installed: 2.2.5]
│       ├── click [required: >=8.0, installed: 8.1.7]
│       ├── importlib-metadata [required: >=3.6.0, installed: 6.11.0]
│       │   └── zipp [required: >=0.5, installed: 3.17.0]
│       ├── itsdangerous [required: >=2.0, installed: 2.1.2]
│       ├── Jinja2 [required: >=3.0, installed: 3.1.3]
│       │   └── MarkupSafe [required: >=2.0, installed: 2.1.3]
│       └── Werkzeug [required: >=2.2.2, installed: 2.2.3]
│           └── MarkupSafe [required: >=2.1.1, installed: 2.1.3]
├── flask-session [required: >=0.4.0, installed: 0.5.0]
│   ├── cachelib [required: Any, installed: 0.9.0]
│   └── Flask [required: >=2.2, installed: 2.2.5]
│       ├── click [required: >=8.0, installed: 8.1.7]
│       ├── importlib-metadata [required: >=3.6.0, installed: 6.11.0]
│       │   └── zipp [required: >=0.5, installed: 3.17.0]
│       ├── itsdangerous [required: >=2.0, installed: 2.1.2]
│       ├── Jinja2 [required: >=3.0, installed: 3.1.3]
│       │   └── MarkupSafe [required: >=2.0, installed: 2.1.3]
│       └── Werkzeug [required: >=2.2.2, installed: 2.2.3]
│           └── MarkupSafe [required: >=2.1.1, installed: 2.1.3]
├── flask-wtf [required: >=0.15, installed: 1.2.1]
│   ├── Flask [required: Any, installed: 2.2.5]
│   │   ├── click [required: >=8.0, installed: 8.1.7]
│   │   ├── importlib-metadata [required: >=3.6.0, installed: 6.11.0]
│   │   │   └── zipp [required: >=0.5, installed: 3.17.0]
│   │   ├── itsdangerous [required: >=2.0, installed: 2.1.2]
│   │   ├── Jinja2 [required: >=3.0, installed: 3.1.3]
│   │   │   └── MarkupSafe [required: >=2.0, installed: 2.1.3]
│   │   └── Werkzeug [required: >=2.2.2, installed: 2.2.3]
│   │       └── MarkupSafe [required: >=2.1.1, installed: 2.1.3]
│   ├── itsdangerous [required: Any, installed: 2.1.2]
│   └── wtforms [required: Any, installed: 3.1.2]
│       └── MarkupSafe [required: Any, installed: 2.1.3]
├── fsspec [required: >=2023.10.0, installed: 2023.12.2]
├── google-re2 [required: >=1.0, installed: 1.1]
├── gunicorn [required: >=20.1.0, installed: 21.2.0]
│   └── packaging [required: Any, installed: 23.2]
├── httpx [required: Any, installed: 0.23.3]
│   ├── certifi [required: Any, installed: 2023.11.17]
│   ├── httpcore [required: >=0.15.0,<0.17.0, installed: 0.16.3]
│   │   ├── anyio [required: >=3.0,<5.0, installed: 4.2.0]
│   │   │   ├── exceptiongroup [required: >=1.0.2, installed: 1.2.0]
│   │   │   ├── idna [required: >=2.8, installed: 3.6]
│   │   │   ├── sniffio [required: >=1.1, installed: 1.3.0]
│   │   │   └── typing-extensions [required: >=4.1, installed: 4.9.0]
│   │   ├── certifi [required: Any, installed: 2023.11.17]
│   │   ├── h11 [required: >=0.13,<0.15, installed: 0.14.0]
│   │   └── sniffio [required: ==1.*, installed: 1.3.0]
│   ├── rfc3986 [required: >=1.3,<2, installed: 1.5.0]
│   └── sniffio [required: Any, installed: 1.3.0]
├── importlib-metadata [required: >=1.7, installed: 6.11.0]
│   └── zipp [required: >=0.5, installed: 3.17.0]
├── importlib-resources [required: >=5.2, installed: 5.13.0]
│   └── zipp [required: >=3.1.0, installed: 3.17.0]
├── itsdangerous [required: >=2.0, installed: 2.1.2]
├── Jinja2 [required: >=3.0.0, installed: 3.1.3]
│   └── MarkupSafe [required: >=2.0, installed: 2.1.3]
├── jsonschema [required: >=4.18.0, installed: 4.21.0]
│   ├── attrs [required: >=22.2.0, installed: 23.2.0]
│   ├── importlib-resources [required: >=1.4.0, installed: 5.13.0]
│   │   └── zipp [required: >=3.1.0, installed: 3.17.0]
│   ├── jsonschema-specifications [required: >=2023.03.6, installed: 2023.12.1]
│   │   ├── importlib-resources [required: >=1.4.0, installed: 5.13.0]
│   │   │   └── zipp [required: >=3.1.0, installed: 3.17.0]
│   │   └── referencing [required: >=0.31.0, installed: 0.31.1]
│   │       ├── attrs [required: >=22.2.0, installed: 23.2.0]
│   │       └── rpds-py [required: >=0.7.0, installed: 0.17.1]
│   ├── pkgutil-resolve-name [required: >=1.3.10, installed: 1.3.10]
│   ├── referencing [required: >=0.28.4, installed: 0.31.1]
│   │   ├── attrs [required: >=22.2.0, installed: 23.2.0]
│   │   └── rpds-py [required: >=0.7.0, installed: 0.17.1]
│   └── rpds-py [required: >=0.7.1, installed: 0.17.1]
├── lazy-object-proxy [required: Any, installed: 1.10.0]
├── linkify-it-py [required: >=2.0.0, installed: 2.0.2]
│   └── uc-micro-py [required: Any, installed: 1.0.2]
├── lockfile [required: >=0.12.2, installed: 0.12.2]
├── Markdown [required: >=3.0, installed: 3.5.2]
│   └── importlib-metadata [required: >=4.4, installed: 6.11.0]
│       └── zipp [required: >=0.5, installed: 3.17.0]
├── markdown-it-py [required: >=2.1.0, installed: 3.0.0]
│   └── mdurl [required: ~=0.1, installed: 0.1.2]
├── MarkupSafe [required: >=1.1.1, installed: 2.1.3]
├── marshmallow-oneofschema [required: >=2.0.1, installed: 3.0.1]
│   └── marshmallow [required: >=3.0.0,<4.0.0, installed: 3.20.2]
│       └── packaging [required: >=17.0, installed: 23.2]
├── mdit-py-plugins [required: >=0.3.0, installed: 0.4.0]
│   └── markdown-it-py [required: >=1.0.0,<4.0.0, installed: 3.0.0]
│       └── mdurl [required: ~=0.1, installed: 0.1.2]
├── opentelemetry-api [required: >=1.15.0, installed: 1.22.0]
│   ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
│   │   └── wrapt [required: >=1.10,<2, installed: 1.16.0]
│   └── importlib-metadata [required: >=6.0,<7.0, installed: 6.11.0]
│       └── zipp [required: >=0.5, installed: 3.17.0]
├── opentelemetry-exporter-otlp [required: Any, installed: 1.22.0]
│   ├── opentelemetry-exporter-otlp-proto-grpc [required: ==1.22.0, installed: 1.22.0]
│   │   ├── backoff [required: >=1.10.0,<3.0.0, installed: 2.2.1]
│   │   ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
│   │   │   └── wrapt [required: >=1.10,<2, installed: 1.16.0]
│   │   ├── googleapis-common-protos [required: ~=1.52, installed: 1.62.0]
│   │   │   └── protobuf [required: >=3.19.5,<5.0.0.dev0,!=4.21.5,!=4.21.4,!=4.21.3,!=4.21.2,!=4.21.1,!=3.20.1,!=3.20.0, installed: 4.25.2]
│   │   ├── grpcio [required: >=1.0.0,<2.0.0, installed: 1.60.0]
│   │   ├── opentelemetry-api [required: ~=1.15, installed: 1.22.0]
│   │   │   ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
│   │   │   │   └── wrapt [required: >=1.10,<2, installed: 1.16.0]
│   │   │   └── importlib-metadata [required: >=6.0,<7.0, installed: 6.11.0]
│   │   │       └── zipp [required: >=0.5, installed: 3.17.0]
│   │   ├── opentelemetry-exporter-otlp-proto-common [required: ==1.22.0, installed: 1.22.0]
│   │   │   ├── backoff [required: >=1.10.0,<3.0.0, installed: 2.2.1]
│   │   │   └── opentelemetry-proto [required: ==1.22.0, installed: 1.22.0]
│   │   │       └── protobuf [required: >=3.19,<5.0, installed: 4.25.2]
│   │   ├── opentelemetry-proto [required: ==1.22.0, installed: 1.22.0]
│   │   │   └── protobuf [required: >=3.19,<5.0, installed: 4.25.2]
│   │   └── opentelemetry-sdk [required: ~=1.22.0, installed: 1.22.0]
│   │       ├── opentelemetry-api [required: ==1.22.0, installed: 1.22.0]
│   │       │   ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
│   │       │   │   └── wrapt [required: >=1.10,<2, installed: 1.16.0]
│   │       │   └── importlib-metadata [required: >=6.0,<7.0, installed: 6.11.0]
│   │       │       └── zipp [required: >=0.5, installed: 3.17.0]
│   │       ├── opentelemetry-semantic-conventions [required: ==0.43b0, installed: 0.43b0]
│   │       └── typing-extensions [required: >=3.7.4, installed: 4.9.0]
│   └── opentelemetry-exporter-otlp-proto-http [required: ==1.22.0, installed: 1.22.0]
│       ├── backoff [required: >=1.10.0,<3.0.0, installed: 2.2.1]
│       ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
│       │   └── wrapt [required: >=1.10,<2, installed: 1.16.0]
│       ├── googleapis-common-protos [required: ~=1.52, installed: 1.62.0]
│       │   └── protobuf [required: >=3.19.5,<5.0.0.dev0,!=4.21.5,!=4.21.4,!=4.21.3,!=4.21.2,!=4.21.1,!=3.20.1,!=3.20.0, installed: 4.25.2]
│       ├── opentelemetry-api [required: ~=1.15, installed: 1.22.0]
│       │   ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
│       │   │   └── wrapt [required: >=1.10,<2, installed: 1.16.0]
│       │   └── importlib-metadata [required: >=6.0,<7.0, installed: 6.11.0]
│       │       └── zipp [required: >=0.5, installed: 3.17.0]
│       ├── opentelemetry-exporter-otlp-proto-common [required: ==1.22.0, installed: 1.22.0]
│       │   ├── backoff [required: >=1.10.0,<3.0.0, installed: 2.2.1]
│       │   └── opentelemetry-proto [required: ==1.22.0, installed: 1.22.0]
│       │       └── protobuf [required: >=3.19,<5.0, installed: 4.25.2]
│       ├── opentelemetry-proto [required: ==1.22.0, installed: 1.22.0]
│       │   └── protobuf [required: >=3.19,<5.0, installed: 4.25.2]
│       ├── opentelemetry-sdk [required: ~=1.22.0, installed: 1.22.0]
│       │   ├── opentelemetry-api [required: ==1.22.0, installed: 1.22.0]
│       │   │   ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
│       │   │   │   └── wrapt [required: >=1.10,<2, installed: 1.16.0]
│       │   │   └── importlib-metadata [required: >=6.0,<7.0, installed: 6.11.0]
│       │   │       └── zipp [required: >=0.5, installed: 3.17.0]
│       │   ├── opentelemetry-semantic-conventions [required: ==0.43b0, installed: 0.43b0]
│       │   └── typing-extensions [required: >=3.7.4, installed: 4.9.0]
│       └── requests [required: ~=2.7, installed: 2.31.0]
│           ├── certifi [required: >=2017.4.17, installed: 2023.11.17]
│           ├── charset-normalizer [required: >=2,<4, installed: 3.3.2]
│           ├── idna [required: >=2.5,<4, installed: 3.6]
│           └── urllib3 [required: >=1.21.1,<3, installed: 1.26.18]
├── packaging [required: >=14.0, installed: 23.2]
├── pathspec [required: >=0.9.0, installed: 0.12.1]
├── pendulum [required: >=2.1.2,<4.0, installed: 3.0.0]
│   ├── backports.zoneinfo [required: >=0.2.1, installed: 0.2.1]
│   ├── importlib-resources [required: >=5.9.0, installed: 5.13.0]
│   │   └── zipp [required: >=3.1.0, installed: 3.17.0]
│   ├── python-dateutil [required: >=2.6, installed: 2.8.2]
│   │   └── six [required: >=1.5, installed: 1.16.0]
│   ├── time-machine [required: >=2.6.0, installed: 2.13.0]
│   │   └── python-dateutil [required: Any, installed: 2.8.2]
│   │       └── six [required: >=1.5, installed: 1.16.0]
│   └── tzdata [required: >=2020.1, installed: 2023.4]
├── pluggy [required: >=1.0, installed: 1.3.0]
├── psutil [required: >=4.2.0, installed: 5.9.7]
├── pydantic [required: >=2.3.0, installed: 2.5.3]
│   ├── annotated-types [required: >=0.4.0, installed: 0.6.0]
│   │   └── typing-extensions [required: >=4.0.0, installed: 4.9.0]
│   ├── pydantic-core [required: ==2.14.6, installed: 2.14.6]
│   │   └── typing-extensions [required: >=4.6.0,!=4.7.0, installed: 4.9.0]
│   └── typing-extensions [required: >=4.6.1, installed: 4.9.0]
├── pygments [required: >=2.0.1, installed: 2.17.2]
├── PyJWT [required: >=2.0.0, installed: 2.8.0]
├── python-daemon [required: >=3.0.0, installed: 3.0.1]
│   ├── docutils [required: Any, installed: 0.16]
│   ├── lockfile [required: >=0.10, installed: 0.12.2]
│   └── setuptools [required: >=62.4.0, installed: 66.1.1]
├── python-dateutil [required: >=2.3, installed: 2.8.2]
│   └── six [required: >=1.5, installed: 1.16.0]
├── python-nvd3 [required: >=0.15.0, installed: 0.15.0]
│   ├── Jinja2 [required: >=2.8, installed: 3.1.3]
│   │   └── MarkupSafe [required: >=2.0, installed: 2.1.3]
│   └── python-slugify [required: >=1.2.5, installed: 8.0.1]
│       └── text-unidecode [required: >=1.3, installed: 1.3]
├── python-slugify [required: >=5.0, installed: 8.0.1]
│   └── text-unidecode [required: >=1.3, installed: 1.3]
├── rfc3339-validator [required: >=0.1.4, installed: 0.1.4]
│   └── six [required: Any, installed: 1.16.0]
├── rich [required: >=12.4.4, installed: 13.7.0]
│   ├── markdown-it-py [required: >=2.2.0, installed: 3.0.0]
│   │   └── mdurl [required: ~=0.1, installed: 0.1.2]
│   ├── pygments [required: >=2.13.0,<3.0.0, installed: 2.17.2]
│   └── typing-extensions [required: >=4.0.0,<5.0, installed: 4.9.0]
├── rich-argparse [required: >=1.0.0, installed: 1.4.0]
│   └── rich [required: >=11.0.0, installed: 13.7.0]
│       ├── markdown-it-py [required: >=2.2.0, installed: 3.0.0]
│       │   └── mdurl [required: ~=0.1, installed: 0.1.2]
│       ├── pygments [required: >=2.13.0,<3.0.0, installed: 2.17.2]
│       └── typing-extensions [required: >=4.0.0,<5.0, installed: 4.9.0]
├── setproctitle [required: >=1.1.8, installed: 1.3.3]
├── SQLAlchemy [required: >=1.4.28,<2.0, installed: 1.4.51]
│   └── greenlet [required: !=0.4.17, installed: 3.0.3]
├── SQLAlchemy-JSONField [required: >=1.0, installed: 1.0.2]
│   └── SQLAlchemy [required: Any, installed: 1.4.51]
│       └── greenlet [required: !=0.4.17, installed: 3.0.3]
├── tabulate [required: >=0.7.5, installed: 0.9.0]
├── tenacity [required: >=6.2.0,!=8.2.0, installed: 8.2.3]
├── termcolor [required: >=1.1.0, installed: 2.4.0]
├── unicodecsv [required: >=0.14.1, installed: 0.14.1]
├── universal-pathlib [required: >=0.1.4, installed: 0.1.4]
│   └── fsspec [required: >=2022.1.0, installed: 2023.12.2]
└── Werkzeug [required: >=2.0,<3, installed: 2.2.3]
    └── MarkupSafe [required: >=2.1.1, installed: 2.1.3]
```
  
</details>

Reply:
Yep, we could. It's not **strictly** necessary (same as packaging in providers is not stricttly necessary) - because of implicit /transitive dependency, but yes, explicit is better than implicit.

Reply:
Added it now.

Reply:
It hard to not  to retrieve `requests` in project like Airflow, in Official docker image it would be always available because we install http provider.

I've also found the trend (or maybe it is only my filling) that projects reduce usage of `requests` in favor or `urllib3` for reduce requests abstraction level around urllib3 and remove some historical behaviour of requests

Reply:
Yes. I would also like to get rid of requests eventually. It's not up-to-date with some recent functionalities like asynchronous requests, HTTP3. - and it's basically in very low maintenance mode (but it's also under PSF umbrella, so at least we can count on security fixes). Friend of mine (creator of charset-normalizer) released https://github.com/jawah/niquests - which have all the bells and whistles for example, but it's hard to depend on quit security fixes releases there and generally betting on maintenance where there are no such many users is not a good idea.

Reply:
I now synced all requests deps to `requests>=2.27.0,<3`.